### PR TITLE
update elixir contrib

### PR DESCRIPTION
**Pull request by tonini**
### Changes

* add missing key bindings
* add alchemist-refcard spacemacs version
* update README

### CODE REVIEW

**Comment by trishume** [comment]
:+1: Not sure how I feel about the unconventional refcard, but it looks nice and might be useful so whatever.

**Comment by tonini** [comment]
haha, indeed it looks definitely not like the GNU standard. But I thought sometimes also the GNU community could use some more shiny things. :smile: 

The thought behind the refcard is that a lot of people that using my Alchemist-Mode and are spacemacs users, asking me about a keybindings overview. And because I can't refer them to my refcard which isn't for spacemacs users at all, It might be a good idea to have a Alchemist-Spacemacs version available.

**Comment by AgIt0** [comment]
IMO as someone who just switched to vim and I have enough overwhelming things to think about I think this is quite helpful. Though I use the readme most of the time it's nice to have it open in something else when you have the extra monitor :+1:

**Comment by syl20bnr** [comment]
@tonini That's great ! :+1: 
We have some [conventions](https://github.com/syl20bnr/spacemacs/blob/master/doc/CONVENTIONS.md) in place though, I'll comment the key bindings that need to be adapted or added.

File: contrib/!lang/erlang-elixir/README.md [hunk 1]
```diff
@@ -98,18 +122,19 @@ Note that `EDTS` is disable on Windows.
 <kbd>SPC m e r</kbd>  | Evaluate region
 <kbd>SPC m e R</kbd>  | Evaluate region and insert result
 
-#### Elixir REPL interactions
+##### REPL interactions
 
     Key Binding       |                 Description
 ----------------------|------------------------------------------------------------
 <kbd>SPC m s i</kbd>  | Start an `iex` inferior process
+<kbd>SPC m s I</kbd>  | Start an IEx process with mix (`iex -S mix`)
```
Comment (syl20bnr) [inline contrib/!lang/erlang-elixir/README.md]: If starting via Mix is the most used way to start an iteractive REPL then we could swap the bindings `SPC m s i` and `SPC m s I`.

File: contrib/!lang/erlang-elixir/README.md [hunk 2]
```diff
@@ -126,5 +160,12 @@ Note that `EDTS` is disable on Windows.
 <kbd>SPC m x b</kbd>  | Run the current buffer through `elixir`
 <kbd>SPC m x f</kbd>  | Run `elixir` with the given filename
 
+##### Code Definition Jump
+
+    Key Binding     |                 Description
+--------------------|------------------------------------------------------------
+<kbd>SPC m .</kbd>  | Jump to the elixir expression definition at point.
+<kbd>SPC m ,</kbd>  | Pop back to where <kbd>SPC m .</kbd> was last invoked.
+
```
Comment (syl20bnr) [inline contrib/!lang/erlang-elixir/README.md]: Add `SPC m g g` to go to definition under point. For now we don't have any conventions to pop back to the previous location though.

File: contrib/!lang/erlang-elixir/README.md [hunk 3]
```diff
@@ -87,7 +102,16 @@ Note that `EDTS` is disable on Windows.
...
-#### Elixir Evaluation in place
+##### Project
+
+    Key Binding       |                 Description
+----------------------|------------------------------------------------------------
+<kbd>SPC m p f</kbd>  | Open project test directory and list all test files.
+<kbd>SPC m p c</kbd>  | Open the appropriate test file for the current buffer file in a new window.
+<kbd>SPC m p t</kbd>  | Toggle between a file and its tests in the current window.
+<kbd>SPC m p T</kbd>  | Toggle between a file and its tests in other window.
+
```
Comment (syl20bnr) [inline contrib/!lang/erlang-elixir/README.md]: This goto should go under `SPC m g`.
`SPC m g t` to go to test
I would use the `SPC m g a` convention (alternate file) to switch back and forth between the tests and the implementation

So my suggestions:
`SPC m p f` ---> `SPC m p t`

`SPC m p t` ---> `SPC m g t`
`SPC m p T` ---> `SPC m g T`

I'm not sure about the usefulness of `SPC m p c`, this should be a fall back when toggling (the first time it is used when the test file is not yet opened).

**Comment by tonini** [comment]
@syl20bnr Thanks for taking the time and giving me feedback about the conventions. :+1: 

> I'm not sure about the usefulness of SPC m p c, this should be a fall back when toggling (the first time it is used when the test file is not yet opened).

You're right, I guess I will also mark this functions as deprecated in the next Alchemist release.

> If starting via Mix is the most used way to start an iteractive REPL then we could swap the bindings SPC m s i and SPC m s I.

That's worth thinking about, but at the moment I think we should let it that way. I would like to ask people who using Alchemist how they use `iex` inside emacs so far.

> Add SPC m g g to go to definition under point. For now we don't have any conventions to pop back to the previous location though.

So would it be ok if we let the `m,` for jumping back for now?

Thanks

**Comment by tonini** [comment]
@syl20bnr Are you ok with the adjustments, or are there any other things we could improve?

**Comment by syl20bnr** [comment]
@tonini I merged it 10 minutes ago, I have not pushed yet ;-)

**Comment by syl20bnr** [comment]
(I don't use the github merge pull request button)

**Comment by tonini** [comment]
@syl20bnr amazing :) Thanks

**Comment by syl20bnr** [comment]
Excellent PR ! :heart: That's wonderful to have the author of a package configure it for spacemacs :+1: 

Thank you !
Cherry-picked into `develop` branch, you can safely delete your branch.

**Comment by tonini** [comment]
@syl20bnr Amazing! Thanks a lot of the constructive review and feedback of the PR and the nice words :heart: 

Thanks you!

**Pull request closed without merging**